statsmodels/sandbox/regression/penalized.py

```diff
@@ -143,33 +334,78 @@ def hatmatrix_diag(self):
 
+    #@cache_readonly
     def hatmatrix_trace(self):
-        return self.hatmatrix_diag().sum()
+        return self.hatmatrix_diag.sum()
 
-    #this doesn't update df_resid
-    @property   #needs to be property or attribute (no call)
-    def df_model(self):
-        return self.hatmatrix_trace()
+##    #this doesn't update df_resid
+##    @property   #needs to be property or attribute (no call)
+##    def df_model(self):
+##        return self.hatmatrix_trace()
 
     #Note: mse_resid uses df_resid not nobs-k_vars, which might differ if df_model, tr(H), is used
     #in paper for gcv ess/nobs is used instead of mse_resid
+    @cache_readonly
     def gcv(self):
         return self.mse_resid / (1. - self.hatmatrix_trace() / self.nobs)**2
 
+    @cache_readonly
     def cv(self):
-        return ((self.resid / (1. - self.hatmatrix_diag()))**2).sum() / self.nobs
+        return ((self.resid / (1. - self.hatmatrix_diag))**2).sum() / self.nobs
 
+    @cache_readonly
     def aicc(self):
         aic = np.log(self.mse_resid) + 1
         aic += 2 * (1. + self.hatmatrix_trace()) / (self.nobs - self.hatmatrix_trace() -2)
         return aic
 
 
+    def test_compatibility(self):
+        """Hypothesis test for the compatibility of prior mean with data
+
+        """
+        # TODO: should we store the OLS results ?  not needed so far, but maybe cache
+        #params_ols = np.linalg.pinv(self.model.exog).dot(self.model.endog)
+        #res = self.wald_test(self.model.r_matrix, q_matrix=self.model.q_matrix, use_f=False)
+        #from scratch
+        res_ols = OLS(self.model.endog, self.model.exog).fit()
+        r_mat = self.model.r_matrix
+        r_diff = self.model.q_matrix - r_mat.dot(res_ols.params)[:,None]
+        ols_cov_r = res_ols.cov_params(r_matrix=r_mat)
+        statistic = r_diff.T.dot(np.linalg.solve(ols_cov_r + self.model.sigma_prior, r_diff))
+        from scipy import stats
+        df = np.linalg.matrix_rank(self.model.sigma_prior)   # same as r_mat.shape[0]
+        pvalue = stats.chi2.sf(statistic, df)
+        # TODO: return results class
+        return statistic, pvalue, df
+
+
+    def share_data(self):
+        """a measure for the fraction of the data in the estimation result
+
+        The share of the prior information is `1 - share_data`.
+
+        Returns
+        -------
+        share : float between 0 and 1
+            share of data defined as the ration between effective degrees of
+            freedom of the model and the number (TODO should be rank) of the
+            explanatory variables.
+
+        """
+
+        # this is hatmatrix_trace / self.exog.shape[1]
+        # This needs to use rank of exog and not shape[1],
+        # since singular exog is allowed
+        return (self.df_model + 1) / self.model.rank  # + 1 is for constant
+
+
 #contrast/restriction matrices, temporary location
 
 def coef_restriction_meandiff(n_coeffs, n_vars=None, position=0):
```

Comment:
docstrings missing
AFAIU, those should be part of penalization tools, i.e. also for other L2 penalization